spec/app_spec.rb

```diff
@@ -0,0 +1,145 @@
+require_relative '../app'
+require 'item_repository'
+require 'order_repository'
+
+RSpec.describe Application do
+  def reset_app_table
+    seed_sql = File.read('spec/orders_seeds.sql')
+    connection = PG.connect({ host: '127.0.0.1', dbname: 'shop_manager_test' })
+    connection.exec(seed_sql)
+  end
+
+  before(:each) do 
+    reset_app_table
+  end
+
+  it "print the header and " do
+    io = double :io
+    order_repository = OrderRepository.new
+    item_repository = ItemRepository.new
+    expect(io).to receive(:puts).with("Welcome to the shop management program!")
+    expect(io).to receive(:puts).with("")
+    expect(io).to receive(:puts).with("What do you want to do?")
+    expect(io).to receive(:puts).with("1 = list all shop items")
+    expect(io).to receive(:puts).with("2 = create a new item")
+    expect(io).to receive(:puts).with("3 = list all orders")
+    expect(io).to receive(:puts).with("4 = create a new order")
+    expect(io).to receive(:gets).and_return("")
+    expect(io).to receive(:puts).with("Please, select a number between 1 and 4")
```

Comment:
One thing i learned from this project is that you could put the lines 17-26 in the before(:each) method and not have to repeat them in each test. This would help to keep the tests looking cleaner and less repitition.